petsctools/options.py

```diff
@@ -210,5 +284,173 @@ def inserted_options(self):
     @functools.cached_property
     def options_object(self):
         from petsc4py import PETSc
-
         return PETSc.Options()
+
+
+def petscobj2str(obj):
+    """Return a string with a PETSc object type and prefix.
+
+    Parameters
+    ----------
+    obj : petsc4py.PETSc.Object
+        The object to stringify.
+
+    Returns
+    -------
+    name : str
+        The stringified name of the object
+    """
+    return f"{type(obj).__name__} ({obj.getOptionsPrefix()})"
+
+
+def attach_options(obj, parameters=None,
+                   options_prefix=None):
+    """Set up an OptionsManager and attach it to a PETSc Object.
+
+    Parameters
+    ----------
+    obj : petsc4py.PETSc.Object
+        The object to attach an OptionsManager to.
+    parameters : Optional[dict]
+        The dictionary of parameters to use.
+    options_prefix: Optional[str]
+        The options prefix to use for this object.
+        See the OptionsManager documentation for more detail.
+
+    Returns
+    -------
+    obj : petsc4py.PETSc.Object
+        The original object.
```

Comment:
Missed this. Also don't return the object here.

Reply:
Shame, I like the possibility of just doing:
```python
set_from_options(attach_options(ksp, parameters))
```

Reply:
I had the idea of having
```
set_from_options(ksp, parameters, prefix)
```
which would call `attach_options` internally and error if it has already been called.